### ISSUE TITLE
Include profile tool in atlas

### ISSUE THREAD

**Issue by cesarcorreo** [qgis/QGIS#37618]
Could be possible to include the profile tool inside the atlas?
Sometimes I'm interested in create and export automated profiles and it could be very interesting to include automated profiles in the atlas in order to create it depending the atlas pages.
Also in order to automate it and to compare it, y-axis should be possible to fix 
Thanks

**Feature description.**

**Additional context**

**Comment by Gustry** [comment]
Are you talking about the profile tool plugin?
If yes, this must be filled against the Profile tool repository, not QGIS.

**Comment by andreasneumann** [comment]
If I may chime in here:
I think that a profile tool should be in the core of QGIS and not rely on plugins anymore. It is a very basic functionality of a GIS to extract raster values along a line. It would also make integration easier in other parts of QGIS (modeler, printing, web services, atlas, reports, etc.)

Or it could be handled by the DataPlotly plugin, which is almost like a "core plugin" and has lots of options for charts and diagrams.

**Comment by gioman** [comment]
> If I may chime in here:
> I think that a profile tool should be in the core of QGIS and not rely on plugins anymore. It is a very basic functionality of a GIS to extract raster values along a line. It would also make integration easier in other parts of QGIS (modeler, printing, web services, atlas, reports, etc.)

@andreasneumann I totally agree with Andreas: while is correct to use/suggest the plugin path for several specialized functionalities I also think that a profile tool should make its way into core QGIS.

I also think that there are at least a couple of other plugins that should become core tools:

value tool
group stats

they provide so common functionalities that they should really become core.

**Comment by roya0045** [comment]
@gioman I think the limiting factor is the fact tat plugin are in python and core stuff are in c++. This mean that either current dev must convert the work in c++ or make something equivalent or have the author learn c++ to re-implement their work. Saying that the work is so common and useful that it should be integrated is a point that I agree. But how to do said integration is more grey.

**Comment by nyalldawson** [comment]
@roya0045 actually this is entirely possible -- the DataPlotly plugin is a good example of a python plugin which extends print layouts by adding a new item type

**Comment by roya0045** [comment]
@nyalldawson I agree that its possible, but since core is in c++ and can be standalone (and more coherent) as opposed to mixing in python, there would be a need to convert the plugin to c++, isn't there? I feel like its stating "You did once in python, now do it again good in c++". Or there would need c++ dev to 'copy' the work of someone else, wich is somewhat dodgy in my view.

**Comment by cesarcorreo** [comment]
Any updated about that possibility?

**Comment by cesarcorreo** [comment]
> Are you talking about the profile tool plugin?
> If yes, this must be filled against the Profile tool repository, not QGIS.

It seems that not any answer in the Profile tool repository. 
I'm still consider that include the Profile tool in the atlas let you prepare automated profiles.
Thanks
